### DISCUSSION TITLE
Codespace is not prepared for an empty repository

### DISCUSSION THREAD

**Discussion by DennesTorres** [community/community#4397] [Codespaces] [answered]
If I create a repository and I intend to use it only with codespaces, it fails. To start using codespaces I need a branch and I end up needing to clone the repository on a client machine and push something in order to create the master branch.

The codespaces UI fails in a strange way when there is no branch in the repo.

It would be interesting if we could start a repo and work with it directly in codespaces.

**Comment by 2percentsilk** [comment, marked as answer]
You can only create a codespace for a repo with at least one file in it - we currently do not support creating a codespace off a completely empty repository. When you create a new repo, be sure to check the box to create a README at least and you should be set!

Sorry for any confusion here!

**Reply by sean-mcclure** [reply]
Then gitpod shouldn't have been removed. Starting a repo from a terminal is outdtated.

**Comment by kallal79** [comment]
Codespace creation failed
Codespace creation failed. Codespace could not be created: Usage not allowed: 7908837174 cannot be billed for new codespaces
